app.ts

```diff
@@ -323,6 +338,94 @@ export default class MELCloudApp extends App {
+  }): Promise<void> {
+    const needsNewCapabilityInstance: boolean = await this.isNewCapabilityInstanceNeeded({ capabilityPath, enabled, threshold })
+    if (!needsNewCapabilityInstance) {
+      return
+    }
+    this.destroyOutdoorTemperatureListener()
+    const [outdoorTemperatureDevice, outdoorTemperatureCapability] = await this.extractOutdoorTemperatureData(capabilityPath)
+    this.outdoorTemperatureListener = outdoorTemperatureDevice.makeCapabilityInstance(
+      outdoorTemperatureCapability,
+      async (): Promise<void> => {
+        try {
+          const [outdoorTemperatureDevice, outdoorTemperatureCapability] = await this.extractOutdoorTemperatureData(capabilityPath)
+          const outdoorTemperature: number = outdoorTemperatureDevice.capabilitiesObj[outdoorTemperatureCapability].value
+          for (const device of this.getDevices({ driverId: 'melcloud' })) {
+            if (device.getCapabilityValue('operation_mode') !== 'cool') {
+              continue
+            }
+            await device.onCapability('target_temperature', Math.max(threshold, outdoorTemperature - 8))
+          }
+        } catch (error: unknown) {
+          this.error(error instanceof Error ? error.message : error)
+          this.destroyOutdoorTemperatureListener(true)
+        }
+      }
+    )
+    this.outdoorTemperatureListener.setValue(enabled).catch(this.error)
+  }
+
+  async isNewCapabilityInstanceNeeded ({ capabilityPath, enabled, threshold }: OutdoorTemperatureListenerData): Promise<boolean> {
+    let haveSettingsChanged: boolean = false
+    if (capabilityPath === '' || threshold === 0) {
+      if (enabled) {
+        throw new Error('Outdoor temperature and/or threshold are missing.')
+      }
+      this.destroyOutdoorTemperatureListener()
+    }
+    if (capabilityPath !== this.homey.settings.get('outdoor_temperature_capability_path')) {
+      this.homey.settings.set('outdoor_temperature_capability_path', capabilityPath)
+      haveSettingsChanged = true
+    }
+    if (threshold !== this.homey.settings.get('self_adjust_threshold')) {
+      this.homey.settings.set('self_adjust_threshold', threshold)
+      haveSettingsChanged = true
+    }
+    if (enabled !== this.homey.settings.get('self_adjust_enabled')) {
+      this.homey.settings.set('self_adjust_enabled', enabled)
+      if (!haveSettingsChanged && this.outdoorTemperatureListener !== null) {
+        this.outdoorTemperatureListener.setValue(enabled).catch(this.error)
+      }
+    }
+    return haveSettingsChanged && capabilityPath !== '' && threshold !== 0
+  }
+
+  destroyOutdoorTemperatureListener (resetData: boolean = false): void {
+    if (resetData) {
+      this.homey.settings.set('outdoor_temperature_capability_path', '')
+      this.homey.settings.set('self_adjust_enabled', false)
+    }
+    if (this.outdoorTemperatureListener !== null) {
+      this.outdoorTemperatureListener.setValue(false).catch(this.error)
+      this.outdoorTemperatureListener.destroy()
```

Comment:
Hi @robertklep, if I well understand HomeyKitty, it's necessary to do `deviceRelatedToOutdoorTemperatureListener.io()` as a workaround in order to make `outdoorTemperatureListener.destroy()` work, is that right?